data_model/src/merkle.rs

```diff
@@ -93,7 +93,7 @@ impl<T> MerkleTree<T> {
     }
```

Comment:
what do you think we move `merkle.rs` into `crypto` library?

Reply:
No preference. But how does the Merkle tree relate to cryptography?

Reply:
`merkle_tree.rs` always seemed like an ugly duckling to me. I would say it relates to cryptography a lot, just open a wiki site on merkle trees and count the number of times word `cryptography` is used. It's not a totally clear categorizational cut.

But there is a pratical reason. I wanted to disable the user from creating raw untyped hashes except out of `HashOf` and this is the one of the two places where `Hash::new` must be used. The other location is somewhat questionable and I'll bring it to attention. So because it's so linked with `Hash` internals I'd like to move it to `iroha_crypto`

Proof in the `view_change.rs` module is quite similar to this situation. Maybe I'd be able to move parts of it to `iroha_crypto` some time, but there is no strong practical reason

Reply:
> I would think that a tree of hashes, is more related to data validation

so is `Hash` and `Signature`, no?

Reply:
Sounds good.